### PR TITLE
CP-16939: fix pvs-farm CLI

### PR DESCRIPTION
**Pull request by robhoes**
Signed-off-by: Rob Hoes rob.hoes@citrix.com

### CODE REVIEW

**Comment by johnelse** [comment]
:+1: :shipit:

File: ocaml/xapi/cli_operations.ml [hunk 1]
```diff
@@ -798,10 +798,12 @@ let gen_cmds rpc session_id =
 		(make_param_funs (Client.GPU_group.get_all) (Client.GPU_group.get_all_records_where) (Client.GPU_group.get_by_uuid) (gpu_group_record) "gpu-group" [] ["uuid";"name-label";"name-description"] rpc session_id) @
 		(make_param_funs (Client.VGPU.get_all) (Client.VGPU.get_all_records_where) (Client.VGPU.get_by_uuid) (vgpu_record) "vgpu" [] ["uuid";"vm-uuid";"device";"gpu-group-uuid"] rpc session_id) @
 		(make_param_funs (Client.VGPU_type.get_all) (Client.VGPU_type.get_all_records_where) (Client.VGPU_type.get_by_uuid) (vgpu_type_record) "vgpu-type" [] ["uuid";"vendor-name";"model-name";"max-resolution";"max-heads"] rpc session_id) @
-		(make_param_funs (Client.DR_task.get_all) (Client.DR_task.get_all_records_where) (Client.DR_task.get_by_uuid) (dr_task_record) "drtask" [] [] rpc session_id)
+		(make_param_funs (Client.DR_task.get_all) (Client.DR_task.get_all_records_where) (Client.DR_task.get_by_uuid) (dr_task_record) "drtask" [] [] rpc session_id) @
```
Comment (lindig) [inline ocaml/xapi/cli_operations.ml:801]: Would it not be better to use `List.concat` here rather than `@` everywhere? I'd be in favour to fix this.

Comment (robhoes) [inline ocaml/xapi/cli_operations.ml:801]: Why would that be better?

**Comment by robhoes** [comment]
Added a patch for bash completion.

File: ocaml/xapi/cli_operations.ml (hunk 1 above)
Comment (lindig) [inline ocaml/xapi/cli_operations.ml:801]: https://gist.github.com/lindig/848dd49c74686432f34537d3ae7eeebf

At least in principle this should be more efficient as List.concat can see all arguments at once rather using a binary operation like `@` that has complexity O(n) in its arguments, leading to quadratic complexity when done repeatedly.

**Comment by lindig** [comment]
Looks good to me.

File: ocaml/xapi/cli_operations.ml (hunk 1 above)
Comment (robhoes) [inline ocaml/xapi/cli_operations.ml:801]: Hmm... I think it would still have to walk though each list to append the next each time...

**Pull request merged**